cpp/src/parquet/encoding.cc

```diff
@@ -416,12 +416,12 @@ void DictEncoderImpl<DType>::WriteDict(uint8_t* buffer) {
 // ByteArray and FLBA already have the dictionary encoded in their data heaps
 template <>
 void DictEncoderImpl<ByteArrayType>::WriteDict(uint8_t* buffer) {
-  memo_table_.VisitValues(0, [&](const ::arrow::util::string_view& v) {
+  memo_table_.VisitValues(0, [&buffer](const ::arrow::util::string_view& v) {
     uint32_t len = static_cast<uint32_t>(v.length());
```

Comment:
should this check for overflow?

Reply:
Overflow of? For now ByteArray elements are capped to int32.